### PR TITLE
Settings support for notifications

### PR DESCRIPTION
**Pull request by BarAgam**

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/notifications/AppNotifier.java [hunk 1]
```diff
@@ -53,41 +60,71 @@ public void notifyNewMessage(ChatMessage chatMessage) {
...
+        .setStyle(createStyle(notificationTitle, notificationSummary, notificationText))
+        .setContentTitle(notificationTitle)
+        .setContentText(notificationText)
+        .setSound(getPreferredRingtoneUri());
+
+    if (shouldVibrate()) {
+      builder.setVibrate(VIBRATION_PATTERN);
+    }
+
+    return builder;
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/notifications/AppNotifier.java]: I liked the extraction process u did here, I think you can make this method return the notification itself and not the builder by using build here.

File: domain/src/main/java/com/teamagam/gimelgimel/domain/utils/PreferencesUtils.java [hunk 2]
```diff
@@ -19,18 +19,20 @@ public boolean isMessageFromSelf(String senderId) {
   }
 
   public boolean shouldUseUtm() {
-    boolean shouldUseUtm = false;
     if (mUserPreferencesRepository.contains(USE_UTM_PREF_KEY)) {
-      shouldUseUtm = mUserPreferencesRepository.getBoolean(USE_UTM_PREF_KEY);
+      return mUserPreferencesRepository.getBoolean(USE_UTM_PREF_KEY);
     }
-    return shouldUseUtm;
+    return false;
```
Comment (teamagam5) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/utils/PreferencesUtils.java:25]: THANKS! I looked for this solution and I was apparently blinded...

**Review by teamagam5 — APPROVED** [review]
We agreed that we are all good with the situation of using the SharedPrefs directly when the UserPrefs are not relevant?

File: app/src/main/java/com/teamagam/gimelgimel/app/notifications/AppNotifier.java (hunk 1 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/notifications/AppNotifier.java]: oh yeah

**Pull request merged**